### DISCUSSION TITLE
A2DP I2S microphone to bluetooth

### DISCUSSION THREAD

**Discussion by Andrewt12** [pschatzmann/ESP32-A2DP#423] [Q&A] [answered]
Hi,

I want to ask if streams-i2s-a2dp example works properly? Im having issues with noise, when I overcome the noise I get distortion and extremely low volume. I've tried 3 i2s mics and 2 esp32's  and the same result. If someone can tell me it works for them without yelling into the mic to get any sound I will persevere.  

This is a great library, I just wish I could make it work for my application.

Thank you all in advance.

**Comment by pschatzmann** [comment, marked as answer]
Please follow the steps described in the Wiki: https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working

You need to learn how to use your own hardware first before you start to do some complex sketches!

All the INMP441 Mems Microphones I was working with have one clean channel and one with noise using 16 bits. That's the reason for the proposed logic to just overwirte the noiy one with the proper one.

This issue can be avoided by reading the data with 32 bits, but then you need to convert the data because A2DP needs 16 bits.
